File: duna-annotations/src/main/java/io/duna/core/service/LocalServices.java
/*
 * Copyright (c) 2016 Duna Open Source Project
 * Ministério do Planejamento, Desenvolvimento de Gestão
 * República Federativa do Brasil
 *
 * This file is part of the Duna Project.
 */
package io.duna.core.service;

import javax.inject.Qualifier;
import java.lang.annotation.*;

/**
 * Annotates an injection point where the local service instances will be injected.
 */
@Qualifier
@Documented
@Target({ElementType.FIELD, ElementType.PARAMETER})
@Retention(RetentionPolicy.RUNTIME)
public @interface LocalServices {
}

File: duna-core/src/main/kotlin/io/duna/core/service/LocalServices.kt
package io.duna.core.service

import javax.inject.Qualifier

/**
 * Annotates an injection point where the local service instances will be injected.
 */
@Qualifier
@MustBeDocumented
@Target(AnnotationTarget.FIELD, AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
annotation class LocalServices
